bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -445,4 +455,35 @@ public boolean isEmpty(SerializerProvider provider, String value) {
         }
     }
 
+    /**
+     * Fetches all the custom properties associated with a given component's instance (including additional custom properties)
+     *
+     * @return {@code Map<String, String>} returns all custom property key value pairs associated with the resource
+     */
+    @NotNull
+    private Map<String, String> getCustomProperties() {
+        Map<String, String> allCustomPropertyPairs = new HashMap<>();
+        Map<String, String> selectedCustomProperties = Optional.ofNullable(this.resource.adaptTo(CustomPropertySelector.class))
+            .map(CustomPropertySelector::getSelectedCustomPropertyPairs)
+            .orElse(null);
+        if (selectedCustomProperties != null) {
+            selectedCustomProperties.forEach(allCustomPropertyPairs::putIfAbsent);
+        }
+        // Additional Custom Properties
+        ValueMap componentResourceMap = this.resource.getValueMap();
+        if (componentResourceMap.containsKey(PN_CUSTOM_PROPERTY_ADDITIONAL_KEYS) && componentResourceMap.containsKey(
+            PN_CUSTOM_PROPERTY_ADDITIONAL_VALUES)) {
```

Comment:
Why is this logic required here ? Ideally core component should not use these property names